doc/source/whatsnew/v2.1.0.rst

```diff
@@ -866,6 +866,7 @@ Other
 - Bug in :meth:`Series.argsort` failing to raise when an invalid ``axis`` is passed (:issue:`54257`)
 - Bug in :meth:`Series.map` when giving a callable to an empty series, the returned series had ``object`` dtype. It now keeps the original dtype (:issue:`52384`)
 - Bug in :meth:`Series.memory_usage` when ``deep=True`` throw an error with Series of objects and the returned value is incorrect, as it does not take into account GC corrections (:issue:`51858`)
+- Bug in :meth:`apply_rule` where timezone data was not propagated when computing holiday observances (:issue:`54580`)
```

Comment:
```suggestion
- Bug in :meth:`apply_rule` where timezone data was not propagated when computing holiday observances (:issue:`54580`)
```
```suggestion
- Bug in :class:`USFederalHolidayCalendar` where timezone data was not propagated when computing holiday observances (:issue:`54580`)
```

Also can you move this to `2.2.0.rst`?

Reply:
Moved it to 2.2.0.rst, changed it to `AbstractHolidayCalendar` since it would have affected anyone defining a custom calendar with a half-open date interval defined like Juneteenth.